xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/main/java/org/xwiki/model/document/DocumentAuthors.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.model.document;
+
+import org.xwiki.stability.Unstable;
+import org.xwiki.user.UserReference;
+
+/**
+ * Define the different authors information that are related to an XWiki Document.
+ *
+ * @version $Id$
+ * @since 13.10RC1
+ */
+@Unstable
+public interface DocumentAuthors
+{
+    /**
+     * The content author is the author of the content field only: this author is used for security reason when a script
+     * needs to be executed.
+     *
+     * @return the reference of the user who authored latest content of the document.
+     */
+    UserReference getContentAuthor();
+
+    /**
+     * The metadata author is the author responsible of any change but the content: saving a new xobject in a document
+     * update the metadata auhor but not the content author.
+     *
+     * @return the reference of the user who authored metadata of the document.
+     */
+    UserReference getMetadataAuthor();
+
+    /**
+     * The displayed author is the "real" author of a document as "the one who impulses the changes". However this
+     * author is only there for display purpose: it should never be used for security reasons. This author is different
+     * from metadata or content author when the document is saved through a script or a mechanism which delegates the
+     * save of the document.
+     *
+     * @return the reference of the user who impulses the creation of the document, without being the real user used
+     *          in the APIs for saving the changes.
+     */
+    UserReference getDisplayedAuthor();
```

Comment:
I hesitated between returning a UserReference or a String here, for handling cases where we want to output a name for the Guest user, but I'm now wondering if it shouldn't be the role of the GuestUserReference to encapsulate that information to have cleaner APIs. WDYT @vmassol?

Reply:
AFAIU `getDisplayedAuthor()` is not about returning the display of the author but something else, and thus we should return a reference. And yes it's the role of GuestUserReference to provide the name (and it does already - through `UserPropertiesResolver`).

Now, I can't say what that something else is since `the user who impulses the creation of the document, without being the real user used in the APIs for saving the changes` is not understandable to me... What does it mean to "impulse the creation of a document". If my boss tells me to create a document in xwiki, does that mean that he's the "display author" (whatever this means)? I think we need a better javadoc and as always, always use examples in the javadoc to make things more clear. BTW from your javadoc, it shouldn't be `getDisplayedAuthor()` but `getImpulseAuthor()`... ;) For me this shows there's a problem in the api name. The fact that it seems hard to document suggests to me there's a problem with this concept ATM and that it might be wrong.

Reply:
>  And yes it's the role of GuestUserReference to provide the name (and it does already - through UserPropertiesResolver).

See https://github.com/xwiki/xwiki-platform/blob/c1bb91f980327e9d7ff05bbc3a7ece78306b1c05/xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-default/src/main/java/org/xwiki/user/internal/GuestConfigurationSource.java#L58

Reply:
> And yes it's the role of GuestUserReference to provide the name (and it does already - through UserPropertiesResolver).

So the snippet you show does not really answer my question: it always will display "Guest". My question was that, for example if I comment as Guest I have the capability to specify a username that will be stored in the comment and used for the display. IMO it would make sense that somehow the GuestUserReference could contain such information for various usecases when we allow Guest to perform actions. I'm still wondering if it would make sense and could be encapsulated in the serialization of the reference itself, or if it should be something specific for each usecase. 
I'm asking this question since it could make sense that a for a document created by Guest, we actually display the name provided by the user in some field. 

> `the user who impulses the creation of the document, without being the real user used in the APIs for saving the changes` is not understandable to me... 

So to clarify the idea here is that a document can be created or saved by a script that has the rights to do so when the user itself does not necessarily have the rights. So the final user who uses the script is the user "who impulses" the creation or save. Easy example is a the usage of change request: the user who performs the changes is not the one who merges, the first one will be this displayed user. 
I agree that the Javadoc can be improved with this and other examples. Now for discussing the concept itself of adding this new API I suggest to do it in my original proposal on the forum: https://forum.xwiki.org/t/introduce-a-xwikidocument-displayedauthor/9365